website_apps_store/controllers/main.py

```diff
@@ -0,0 +1,197 @@
+                     'ilike', srch),
+                    ('product_variant_ids.app_description_rst_html',
+                     'ilike', srch),
+                    ('product_variant_ids.app_author_ids.name', 'ilike', srch),
+                    ('product_variant_ids.app_summary', 'ilike', srch)]
+        if category:
+            domain += [('public_categ_ids', 'child_of', int(category))]
+
+        if attrib_values:
+            attrib = None
+            ids = []
+            for value in attrib_values:
+                if not attrib:
+                    attrib = value[0]
+                    ids.append(value[1])
+                elif value[0] == attrib:
+                    ids.append(value[1])
+                else:
+                    domain += [('attribute_line_ids.value_ids', 'in', ids)]
+                    attrib = value[0]
+                    ids = [value[1]]
+            if attrib:
+                domain += [('attribute_line_ids.value_ids', 'in', ids)]
+
+        return domain
+
+    @http.route([
+        '/shop',
+        '/shop/page/<int:page>',
+        '/shop/category/<model("product.public.category"):category>',
+        '/shop/category/<model("product.public.category"):category>\
+        /page/<int:page>'
+    ], type='http', auth="public", website=True)
+    def shop(self, page=0, category=None, search='', ppg=False, **post):
+        res = super(WebsiteSaleCustom, self).shop(
+            page=page, category=category, search=search, ppg=ppg, **post)
+
+        if ppg:
+            try:
+                ppg = int(ppg)
+            except ValueError:
+                ppg = PPG
+            post["ppg"] = ppg
+        else:
+            ppg = PPG
+
+        attrib_list = request.httprequest.args.getlist('attrib')
+        attrib_values = [
+            [int(x) for x in v.split("-")] for v in attrib_list if v]
+        attributes_ids = {v[0] for v in attrib_values}
+        attrib_set = {v[1] for v in attrib_values}
+        domain = self._get_search_domain(search, category, attrib_values)
+
+        keep = QueryURL('/shop', category=category and int(category),
+                        search=search, attrib=attrib_list,
+                        order=post.get('order'),
+                        version=post.get('version'), author=post.get('author'))
+        if post.get('version'):
+            domain += [('product_variant_ids.attribute_value_ids.id',
+                        '=', post.get('version'))]
+        if post.get('author'):
+            domain += [('product_variant_ids.app_author_ids.id',
+                        '=', post.get('author'))]
+
+        url = "/shop"
+        if category:
+            category = request.env['product.public.category'].browse(
+                int(category))
+            url = "/shop/category/%s" % slug(category)
+
+        category_all = request.env['product.public.category'].search([])
+        versions = request.env['product.attribute.value'].search([])
```

Comment:
Shouldn't this be filtered by `attribute_id =  ref('apps_product_creator.attribute_odoo_version')` ? If an other `product.attribute` was created it would be confusing. `product.attribute` can be a lot of things.